pkg/refactor/filestore/psql/root.go

```diff
@@ -0,0 +1,122 @@
+package psql
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/direktiv/direktiv/pkg/refactor/filestore"
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+type Root struct {
+	ID uuid.UUID
+
+	Files []File `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
+
+	CreatedAt time.Time
+	UpdatedAt time.Time
+
+	db *gorm.DB
+}
+
+var _ filestore.Root = &Root{}
+
+type RootList []*Root
+
+func (r *Root) GetID() uuid.UUID {
+	return r.ID
+}
+
+func (r *Root) Delete(ctx context.Context) error {
+	res := r.db.WithContext(ctx).Delete(&Root{ID: r.ID})
+	if res.Error != nil {
+		return res.Error
+	}
+	if res.RowsAffected != 1 {
+		return fmt.Errorf("unexpedted gorm delete count, got: %d, want: %d", res.RowsAffected, 1)
+	}
+
+	return nil
+}
+
+//nolint:ireturn
+func (r *Root) CreateFile(ctx context.Context, path string, typ filestore.FileType, dataReader io.Reader) (filestore.File, error) {
+	path = filepath.Clean(path)
```

Comment:
Let's put an extra check/step in here to either ensure absolute paths or convert relative paths to absolute ones. That will reduce the number of possible bugs related to filepaths.